src/lib/components/Swap/Settings/MaxSlippageSelect.tsx

```diff
@@ -78,8 +78,8 @@ export default function MaxSlippageSelect() {
         <InputOption value={custom} onSelect={onInputSelect} selected={maxSlippage === CUSTOM}>
           <DecimalInput
             size={custom === undefined ? undefined : 5}
-            value={custom}
-            onChange={(custom) => setMaxSlippage({ value: CUSTOM, custom })}
+            value={custom?.toString()}
+            onChange={(custom) => (custom ? setMaxSlippage({ value: CUSTOM, custom: parseFloat(custom) }) : null)}
```

Comment:
Unless you've changed `setMaxSlippage`, I think we want this to always invoke it, with `custom: custom ? parseFloat(custom) : null`.

Reply:
should we assign back to undefined instead? -> custom ? parseFloat(custom) : undefined

Reply:
Let's do `custom ? parseFloat(custom) : undefined`. It's logically identical to `custom && parseFloat(custom)`, but i think it's more clear to a reader.